pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
-        else:
-            single_gpu_ordinal = device_parser.determine_root_gpu_device(self.parallel_device_ids)
-            plugin = SingleDeviceStrategy(device=single_gpu_ordinal if self.use_gpu else "cpu")
-        return plugin
-
-    def resolve_strategy(self, training_type: Strategy) -> Strategy:
-        # necessary for when the user has passed in a plugin
-        if hasattr(training_type, "parallel_devices") and getattr(training_type, "parallel_devices") is None:
-            training_type.parallel_devices = self.parallel_devices
-            if hasattr(training_type, "num_processes"):
-                training_type.num_processes = len(self.parallel_devices)
-
-        if hasattr(training_type, "cluster_environment") and getattr(training_type, "cluster_environment") is None:
-            # transfer ownership of the cluster environment to the training type
-            training_type.cluster_environment = self.cluster_environment
-            self._cluster_environment = proxy(self.cluster_environment)
-
-        if hasattr(training_type, "num_nodes"):
-            # set num_nodes for training_type from trainer setting
-            training_type.num_nodes = self.num_nodes
-
-        if hasattr(training_type, "sync_batchnorm"):
-            # set sync_batchnorm for training_type from trainer setting
-            training_type.sync_batchnorm = self.sync_batchnorm
-
-        return training_type
-
-    def select_accelerator(self) -> Accelerator:
-        if isinstance(self.distributed_backend, Accelerator):
-            # custom accelerator from user
-            if self._precision_plugin is not None or self._strategy is not None:
-                # plugins also specified by user
-                rank_zero_warn(
-                    "Specified `Precision` and `TrainingType` plugins will be ignored,"
-                    " since an `Accelerator` instance was provided."
+        if (
+            self._precision_flag == 16
+            and isinstance(self.accelerator, CPUAccelerator)
+            and self._amp_type_flag == AMPType.APEX
+        ):
+            raise MisconfigurationException(
+                "You passed `Trainer(accelerator='cpu', precision=16, amp_type='apex')`"
+                " but apex AMP not supported on CPU."
+            )
+        if self._precision_flag == "bf16" and self._amp_type_flag != AMPType.NATIVE:
+            raise MisconfigurationException(
+                f"You passed `Trainer(amp_type={self._amp_type_flag.value!r}, precision='bf16')` but "
+                "it's not supported. Try using `amp_type='native'` instead."
+            )
+        if self._precision_flag in (16, "bf16") and self._amp_type_flag == AMPType.APEX:
+            if isinstance(self.strategy, (DDPShardedStrategy, DDPSpawnShardedStrategy, DDPFullyShardedStrategy)):
+                raise MisconfigurationException(
+                    "Sharded plugins are not supported with apex, please switch to `amp_backend='native'`."
```

Comment:
this too can be moved to the strategies

Reply:
I agree we should move as many checks to to individual class as possible. How about in the follow up PRs? 
I don't think we should include in this PR because (a.) it broke the bunch of current tests' error message order (b.) Moving some of the checks requires changes in the class API. For example, precisionPlugins doesn't have precision_flag(16/32), and we need to add it. If we move the check to the class, the PR will be huge

Reply:
Tracked in #11449 follow up item 1